google/cloud/bigtable/testing/table_integration_test.cc

```diff
@@ -99,11 +101,11 @@ std::string TableTestEnvironment::CreateRandomId(std::string const& prefix,
 std::string TableTestEnvironment::RandomTableId() {
   // This value was discovered by trial and error, it is not documented in the
   // proto files.
-  constexpr int kMaxTableIdLength = 50;
-  static char const prefix[] = "table-";
-  static_assert(kMaxTableIdLength > sizeof(prefix), "prefix is too long");
-  constexpr int sample_count = kMaxTableIdLength - sizeof(prefix) + 1;
-  return CreateRandomId(prefix, sample_count);
+  auto constexpr kMaxTableIdLength = 50;
+  static char const kPrefix[] = "table-";  // NOLINT(modernize-avoid-c-arrays)
```

Comment:
(No change needed) I hate this clang-tidy check (modernize-avoid-c-arrays). I'd be happy to disable it.

Alternatively, a good way to avoid it is to do something like: `static const auto kPrefix = "foo"`, which, IIRC, fools it.

Reply:
I think I tried that, and it makes `kPrefix` a `char const*`, which does not work if you use `sizeof()` or any static test on the length of the string (as we do here).